packages/snaps-rpc-methods/src/permitted/createInterface.ts

```diff
@@ -0,0 +1,116 @@
+import type { JsonRpcEngineEndCallback } from '@metamask/json-rpc-engine';
+import type { PermittedHandlerExport } from '@metamask/permission-controller';
+import { rpcErrors } from '@metamask/rpc-errors';
+import type {
+  Component,
+  CreateInterfaceParams,
+  CreateInterfaceResult,
+  JsonRpcRequest,
+} from '@metamask/snaps-sdk';
+import { ComponentStruct, assert } from '@metamask/snaps-sdk';
+import type { InferMatching } from '@metamask/snaps-utils';
+import type { PendingJsonRpcResponse } from '@metamask/utils';
+import { StructError, create, object } from 'superstruct';
+
+import type { MethodHooksObject } from '../utils';
+
+const hookNames: MethodHooksObject<CreateInterfaceMethodHooks> = {
+  hasPermission: true,
+  createInterface: true,
+};
+
+export type CreateInterfaceMethodHooks = {
+  /**
+   * @param origin - The origin invoking the rpc-method.
+   * @param permissionName - The name of the permission invoked.
+   * @returns Whether if the snap has permission to invoke this rpc-method.
+   */
```

Comment:
```suggestion
  /**
   * @param origin - The origin invoking the RPC method.
   * @param permissionName - The name of the permission invoked.
   * @returns Whether the snap has permission to invoke this RPC method or not.
   */
```

Applies to all the RPC methods here